src/main/java/com/fasterxml/jackson/databind/deser/std/FromStringDeserializer.java

```diff
@@ -395,6 +401,45 @@ protected int _firstHyphenOrUnderscore(String str)
             }
             return -1;
         }
+
+        private Locale _deSerializeBCP47Locale(String value, int ix, String first, String second) {
+            String third = "";
+            try {
+                int scriptExpIx = value.indexOf("_#");
+                /*
+                 * Below condition checks if variant value is present to handle empty variant values such as
+                 * en__#Latn_x-ext
+                 * _US_#Latn
+                 */
+                if (scriptExpIx > 0 && scriptExpIx > ix)
+                    third = value.substring(ix + 1, scriptExpIx);
+                value = value.substring(scriptExpIx + 2);
+
+                if (value.indexOf('_') < 0 && value.indexOf('-') < 0) {
+                    return new Locale.Builder().setLanguage(first)
+                            .setRegion(second).setVariant(third).setScript(value).build();
+                }
+                if (value.indexOf('_') < 0) {
+                    ix = value.indexOf('-');
+                    return new Locale.Builder().setLanguage(first)
+                            .setRegion(second).setVariant(third)
+                            .setExtension(value.charAt(0), value.substring(ix + 1))
+                            .build();
+                }
+                ix = value.indexOf('_');
+                return new Locale.Builder().setLanguage(first)
+                        .setRegion(second).setVariant(third)
+                        .setScript(value.substring(0, ix))
+                        .setExtension(value.charAt(ix + 1), value.substring(ix + 3))
+                        .build();
+            } catch(IllformedLocaleException ex) {
+                return new Locale(first, second, third);
+            }
+        }
+
+        private boolean _isScriptOrExtensionPresent(String value) {
+            return value.contains("_#");
```

Comment:
Is this separator/marker documented somewhere? Would be great to have a link to explanation.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#toString()